src/Idris/Doc/String.idr

```diff
@@ -347,9 +347,8 @@ getDocsForName fc n config
                 case !(traverse (pterm . map defaultKindedName) (parents iface)) of
                      [] => []
                      ps => [hsep (header "Constraints" :: punctuate comma (map (prettyBy Syntax) ps))]
-             let icon = case dropNS (iconstructor iface) of
-                          DN _ _ => [] -- machine inserted
-                          nm => [hsep [header "Constructor", dCon nm (prettyName nm)]]
+             icon <- do doc <- getDConDoc =<< toFullNames (iconstructor iface)
+                        pure $ [header "Constructor" <++> annotate Declarations doc]
```

Comment:
I think that here we want to keep the logic that we're only showing the
constructors that have been user-declared, not the machine-introduced
ones.

Otherwise we may give users the wrong idea: these `__mkInterface` names
are not actually valid, cf.

```idris
interface Pointed a where
  point : a

failing "Undefined name mkPointed"

  nat : Pointed Nat
  nat = __mkPointed 0
```

Additionally, I don't think we should use: `getDConDoc` because that'll give
us the type too but here the type is not interesting: we are essentially defining
a record and so the type of the constructor is obvious from the type of the
methods IMO.